src/sentry/static/sentry/app/components/events/interfaces/frame.jsx

```diff
@@ -93,6 +93,19 @@ const Frame = createReactClass({
     };
   },
 
+  componentDidMount() {
+    //Need to update if the SentryAppComponentsStore changes
+    this.unlistenSentryComponentStore = SentryAppComponentsStore.listen(() =>
+      this.forceUpdate()
```

Comment:
This and what we're doing in `getSentryAppComponents()` is not ideal since it kind of breaks how React works. Since these app components are not held in state or props  (and instead we directly access the store for it), we'll be unable to do any optimizations around it if it were ever unnecessary.

I think better would be to have a higher-order component handle feeding `Frame` a list of components from store. You could then have an interface like:

```
export default withSentryAppComponents(<type>)(Frame);
```

Reply:
@billyvg Would I couple that with a `mixin`such as in `withOrganizations.jsx`?

Reply:
@scefali yeah, or `Reflux.listenTo` like in `withTeams` (or you can employ the same `listen()` in `DidMount` like in this component if you wanted to avoid mixins). I think either way, having the HoC will be nice to have that logic out of the Frame component.

Reply:
@billyvg Made the change based on your suggestion